doc/developers/maintainer.rst.template

```diff
@@ -0,0 +1,385 @@
+      target the next version can contribute their changelog entries to this file in
+      parallel to the release process.
+    {% endif %}
+
+    - In the `{{ version_main }}.X` branch, change the version number `__version__` in
+      `sklearn/__init__.py` to `{{ version_full }}`.
+
+    {% if key != "rc" %}
+    - In the `main` branch, edit the corresponding file in the `doc/whats_new` directory
+      to update the release date
+      {%- if key == "final" %}, link the release highlights example,{% endif %}
+      and add the list of contributor names. Suppose that the tag of the last release in
+      the previous major/minor version is `{{ previous_tag }}`, then you can use the
+      following command to retrieve the list of contributor names:
+
+      .. prompt:: bash
+
+        git shortlog -s {{ previous_tag }}.. |
+          cut -f2- |
+          sort --ignore-case |
+          tr "\n" ";" |
+          sed "s/;/, /g;s/, $//" |
+          fold -s
+
+      Then cherry-pick it in the `{{ version_main }}.X` release branch.
+
+    - In the `main` branch, edit `doc/templates/index.html` to change the "News" section
+      in the landing page, along with the month of the release.
+      {%- if key == "final" %}
+      Do not forget to remove old entries (two years or three releases ago) and update
+      the "On-going development" entry.
+      {%- endif %}
+      Then cherry-pick it in the `{{ version_main }}.X` release branch.
+    {% endif %}
+
+    - Trigger the wheel builder with the `[cd build]` commit marker. See also the
+      `workflow runs of the wheel builder
+      <https://github.com/scikit-learn/scikit-learn/actions/workflows/wheels.yml>`_.
+
+      .. prompt:: bash
+
+        git commit --allow-empty -m "[cd build] Trigger wheel builder workflow"
+
+      .. note::
+
+        The acronym CD in `[cd build]` stands for `Continuous Delivery
+        <https://en.wikipedia.org/wiki/Continuous_delivery>`_ and refers to the
+        automation used to generate the release artifacts (binary and source
+        packages). This can be seen as an extension to CI which stands for `Continuous
+        Integration <https://en.wikipedia.org/wiki/Continuous_integration>`_. The CD
+        workflow on GitHub Actions is also used to automatically create nightly builds
+        and publish packages for the development branch of scikit-learn. See also
+        :ref:`install_nightly_builds`.
+
+    - Once all the CD jobs have completed successfully in the PR, merge it again with
+      the `[cd build]` marker in the commit message. This time the results will be
+      uploaded to the staging area. You should then be able to upload the generated
+      artifacts (`.tar.gz` and `.whl` files) to https://test.pypi.org/ using the "Run
+      workflow" form for the `PyPI publishing workflow
+      <https://github.com/scikit-learn/scikit-learn/actions/workflows/publish_pypi.yml>`_.
```

Comment:
```suggestion
    - Once all the CD jobs have completed successfully in the PR, merge it with the
      `[cd build]` marker in the commit message. This time the results will be
      uploaded to the staging area. You should then be able to upload the generated
      artifacts (`.tar.gz` and `.whl` files) to https://test.pypi.org/ using the "Run
      workflow" form for the `PyPI publishing workflow
      <https://github.com/scikit-learn/scikit-learn/actions/workflows/publish_pypi.yml>`_.

      .. warning::
    
        This PR should be merged with the rebase mode instead of the usual squash mode
        because we want to keep the history in the `{{ version_main }}.X` branch close
        to the history of the main branch which will help for future bug fix releases.
          
        In addition if on merging, the last commit, containing the `[cd build]` marker,
        is empty, the CD jobs won't be triggered. In this case, you can directly push
        a commit with the marker in the `{{ version_main }}.X` to trigger them.
```